subiquity/server/controllers/source.py

```diff
@@ -176,3 +179,49 @@ async def POST(self, source_id: str, search_drivers: bool = False) -> None:
 
         if changed or not self._configured:
             await self.configured()
+
+    # The identity section (and user-data) section are entirely optional
+    # on Desktop, but setting the variant doesn't happen until controllers
+    # are started. For the identity controller to properly validate
+    # this behavior, we need to do look-ahead on the source controller
+    # logic for loading the available sources.
+    # See: https://github.com/canonical/subiquity/pull/1965
+    def _install_source_is_desktop(self) -> bool:
+        # Check the autoinstall config first
+        source_config = self.app.autoinstall_config.get(self.autoinstall_key)
+        if (
+            source_config is not None
+            and (id := source_config.get("id")) is not None
```

Comment:
I think I'm vetoing this approach, as it relies on behavior in a string that is theoretically an arbitrary value.
What I want instead is for the `SourceController.start` logic to move earlier in the sequence - out of `start` entirely actually - so that we can answer this question before `SourceController.start()` has been called.  Then we should be able to just get the info needed.  What do you think?

Reply:
When I originally looked at this bug I think I had reservations about moving the logic in `SourceController.start` to somewhere sooner, but looking at it again now I don't see why we can't. We can probably move that logic to `SourceController.load_autoinstall_data`. I'll give it a shot.